### ISSUE TITLE
Downloading or viewing the PDF report is not possible on Andriod phone.

### ISSUE THREAD

**Issue by elainetao829** [d33pspace/RenewalWebsite#23]
Steps:
1. User sends https://staging.renewal365.org to a WeChat account.
2. User clicks the link in WeChat with an Android phone.
3. User opens page account management.
4. User opens tab History.
5. User clicks PDF and waits.

Actual Result:
The page starts loading something, but nothing happens after loading finishes.

Expected Result:
If downloading PDF function is available on cellphone, please fix this bug.
If that function will not be available, please remove this button or disable it. It's confusing.

**Comment by d33pspace** [comment]
Thanks. Can you try this on iOS as well before I turn this over to Girish?

**Comment by d33pspace** [comment]
@kolte Could you please check this out? It seems the PDF is not downloading on Android devices.

**Comment by kolte** [comment]
I just tried this from my android phone and I was able to download, can you please confirm which android phone you are using? and which browser? @d33pspace

**Comment by d33pspace** [comment]
Hi @kolte I was going to check this out on my Pixel but it seems that the PDF button is not working at all now after the last update, including desktop browsers (at lest Edge).

**Comment by d33pspace** [comment]
Hi Elaine, does the Android phone you tried downloading the PDF on happen to be a Xiaomi? I tried downloading on my Xiaomi and it didn't work, but it did work on my Pixel. Girish said it also worked on his Android phone. Just trying to find the common denominator here.

**Comment by elainetao829** [comment]
Yes it's Xiaomi. And it happens to my two Xiaomi phones (xiaomi 6 and mix 2s).
I will find a Huawei and try that function tonight,

**Comment by d33pspace** [comment]
Yes, please try with a Huawei if you can. Also on Xiaomi, try both the standard Xiaomi browser and also from within the WeChat browser.

**Comment by d33pspace** [comment]
@elainetao829 One more question before I pass this off to Girish. The screenshot above may be a hint for us . . . I wonder if this is a problem with the Chinese file name. Have you tried setting to English language and downloading the PDF? Is the problem the same? You can try and confirm on just one phone for this. Thanks!

**Comment by elainetao829** [comment]
Dear @d33pspace,
Thanks for your observation. I just tried on Xiaomi in English. But the PDF still couldn't be downloaded.

**Comment by kolte** [comment]
@d33pspace can you please prepare video for this and send me?
It should just work as we are using standard filestream to download the file. If I can see what happen I will be able to identify the issue.

**Comment by d33pspace** [comment]
Sorry this is still a problem. Before you made the change today, the PDF filenames that my Xiaomi used were:

ENGLISH: `Renewal giving record.pdf` (This file downloaded directly with no problem.)
CHINESE: `________.pdf; filename=UTF-8"日新中心捐赠记录.pdf` (File name prompt.)

After you made the change, the PDF file names are:

English: `Renewal_2018-09-14.pdf; filename=UTF-8"Renewal_2018-09-14.pdf` (File name prompt.)
Chinese: `Rixin_2018-09-14.pdf; filename=UTF-8"Rixin_2018-09-14.pdf` (File name prompt.)

An example of the filename prompt is the suggested file name:
`Rixin_2018-09-14.pdf; filename*=UTF-8"Rixin_2018-09-14.pdf`
Then when you save it says an illegal character was removed from the filename. Note it's the asterisk (*) that is removed.

I tried simplifying the file name to simply Renewal.pdf in our code and I tested that but I'm still getting the file name prompt in both English and Chinese in the Xiaomi browser. 

Does this help you solve this? I can't figure out why before you made the change today that the English PDF downloaded with no problem. If it's a problem of fancy file names we can just use "Renewal.pdf". I've also noticed some sites prompt the user for a file name and location every time for PDF download. This would also be fine in our case if it would prevent these problems with the Chinese phones.
